packages/fxa-settings/src/components/ConnectedServices/Service.tsx

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { LinkExternal } from 'fxa-react/components/LinkExternal';
+import { DeviceLocation } from '../../models/Account';
+import { ReactComponent as WebIcon } from './web.svg';
+import { ReactComponent as DesktopIcon } from './desktop.svg';
+import { ReactComponent as TabletIcon } from './tablet.svg';
+import { ReactComponent as MobileIcon } from './mobile.svg';
+import { ReactComponent as LockwiseIcon } from './lockwise.svg';
+import { ReactComponent as PocketIcon } from './pocket.svg';
+import { ReactComponent as MonitorIcon } from './monitor.svg';
+import { ReactComponent as SyncIcon } from './sync.svg';
+import { ReactComponent as FPNIcon } from './fpn.svg';
+
+export function Service({
+  name,
+  deviceType,
+  location,
+  lastAccessTimeFormatted,
+  canSignOut,
+}: {
+  name: string;
+  deviceType: string;
+  location: DeviceLocation;
+  lastAccessTimeFormatted: string;
+  canSignOut: boolean;
+}) {
+  const { city, stateCode, country } = location;
+  const locationProvided = Boolean(city && stateCode && country);
+  const isPocket = name === 'Pocket';
+  const isMonitor = name === 'Firefox Monitor';
+  const isLockwise = name === 'Firefox Lockwise';
+  const isSync = name === 'Firefox Sync';
+  const isFPN = name === 'Firefox Private Network';
+
+  const renderLink = isPocket || isMonitor || isLockwise || isSync || isFPN;
+  let serviceLink = '';
+  let iconSvg = <WebIcon data-testid="web-icon" />;
+
+  if (renderLink) {
+    if (isPocket) {
+      serviceLink = 'https://www.mozilla.org/en-US/firefox/pocket/';
+      iconSvg = <PocketIcon data-testid="pocket-icon" />;
+    }
+    if (isMonitor) {
+      serviceLink = 'https://monitor.firefox.com/';
+      iconSvg = <MonitorIcon data-testid="monitor-icon" />;
+    }
+    if (isLockwise) {
+      serviceLink = 'https://www.mozilla.org/en-US/firefox/lockwise/';
+      iconSvg = <LockwiseIcon data-testid="lockwise-icon" />;
+    }
+    if (isSync) {
+      serviceLink =
+        'https://support.mozilla.org/en-US/kb/how-do-i-set-sync-my-computer';
+      iconSvg = <SyncIcon data-testid="sync-icon" />;
+    }
+    if (isFPN) {
+      serviceLink = 'https://vpn.mozilla.com/';
+      iconSvg = <FPNIcon data-testid="fpn-icon" />;
+    }
+  } else {
+    const lowerName = name.toLowerCase();
+    if (deviceType === 'mobile') {
+      iconSvg = <MobileIcon data-testid="mobile-icon" />;
+    }
+    if (lowerName.includes('ipad')) {
+      iconSvg = <TabletIcon data-testid="tablet-icon" />;
+    }
+    if (deviceType === 'desktop') {
+      iconSvg = <DesktopIcon data-testid="desktop-icon" />;
+    }
+  }
```

Comment:
I don't know the likelihood of this happening, but each `if` statement from L44-64 and L67-75 is evaluated independently because `else`s are not used, which means more than one could evaluate to `true` and result in undesired output. As well, with such a large number of conditionals you might be better off using a switch case and matching against `true` instead of if/else here.

Reply:
The way these are ordered is on purpose, if mobile and ipad, we want to show the ipad logo. There should never be device type desktop with ipad as the device name. If we see that then that would expose an issue with either the way we are normalizing data in graphql or how we are collecting the device data.

Reply:
As for the difference between 44-67 (services) and 67-75(devices). the else statement means that 67-75 should never be hit by something that requires a service link. We never want them to be overidden by a device type icon, as the service Icon is more specific.

Reply:
You could still use a switch case with a fall through, but this is fine.

Reply:
> There should never be device type desktop with ipad as the device name. If we see that then that would expose an issue with either the way we are normalizing data in graphql or how we are collecting the device data.

How are we monitoring for this? Maybe it should raise an error that sends it to Sentry?

Reply:
I'm not sure, I haven't seen any devices or stub data that is shaped in this way. Worst case scenario is sounds like is that it would just have the web icon. I think I'm not understanding the ask here. I'm happy to add error handling, just wanna make sure I get what case we are handling.

Reply:
Awesome! My suggestion is mostly based in the idea that we're displaying these services based on the string comparisons (does the name equal this, does the device type equal that), and while we may know all of the current values today, they could change down the road. I agree that the web icon should be the fallback if nothing matches, I was mostly suggesting we raise an exception under the hood if none of the currently known services are matched, so we could be proactive about updating instead of waiting for someone to report to us that something doesn't look right.

That being said, it's not critical to do today. What I would like to see down the road is maybe using constants here instead of strings, and one central frontend spot where we could review all possible services: https://jira.mozilla.com/browse/FXA-2607